CovertInfiltration/Src/CovertInfiltration/Classes/X2ActivityTemplate_Assault.uc

```diff
@@ -76,6 +78,21 @@ static function CreateMission (XComGameState NewGameState, XComGameState_Activit
 		/* bUseSpecifiedLevelSeed */, /* LevelSeedOverride */, false /* bSetMissionData */
 	);
 
+	// If the chain has a dark event attached, and this mission has a dark event reward, connect the two
+	if (ActivityState.GetActivityChain().GetChainDarkEvent() != none)
+	{
+		`CI_Log("mission found");
+		foreach MissionState.Rewards(RewardRef)
+		{
+			RewardState = XComGameState_Reward(`XCOMHISTORY.GetGameStateForObjectID(RewardRef.ObjectID));
+			if (RewardState.GetMyTemplateName() == 'Reward_DarkEvent')
+			{
+				`CI_Log("reward found");
+				RewardState.SetReward(ActivityState.GetActivityChain().GetChainDarkEvent().GetReference());
+			}
+		}
+	}
```

Comment:
I would rather if this was done in `X2ActivityTemplate_Mission::InitializeMissionRewards`. This is too hardcoded for my taste